ls1intum/Artemis
# Prepare exercises in parallel, download repos in parallel add option to only prepare exams.

### Checklist

### Motivation and Context
The exam preparation currently happens in only one thread. As we have to wait for external systems like Bitbucket and Bamboo quite long, the exam generation takes around 3-5s per generated programming exercise. Especially for large exams, this is bad, as the exam preparation takes very long.
Therefore we want to speed up the preparation by using parallel execution.

### Description
We execute the exam preparation in parallel using `parallelStream()`. This uses `NUMBER_OF_CPUS - 1` threads instead of `1` thread, therefore the preparation is faster.
We tested it on TS1 and got 2mins instead of 5mins21secs for 100 exams with 1 programming, quiz, text and modelling exercise each.

Also the K6 exam simulation is updated to support only the creation of the exam without preparation of the exercises if the `-op` flag is used.

### Steps for Testing
1. Create an exam with programming exercises
2. Add students (more students for better comparison)
3. The generation should faster, depending on the number of CPUs used in the system.
